### ISSUE TITLE
Category subclasses should have restrictions on the types of things they categorize

### ISSUE THREAD

**Issue by rjyounes** [semanticarts/gist#519]
E.g., `EquipmentType` categorizes `Equipment`:

```
	owl:intersectionOf (
		gist:Category 
		[
			a owl:Restriction ;
		         owl:onProperty [
			    owl:inverseOf gist:categorizedBy ;
			]
		        owl:allValuesFrom gist:Equipment ;
		]
	) ;
```

Might be a good use case for defining the inverse `categorizes` to avoid the repetition of the blank node inverse.

**Comment by uscholdm** [comment]
Yes we could do this sort of thing, across the board and I have thought of doing so.  In the end, while correct, it is not really shedding much light, and its sort of obvious..  I also don't see it being very useful to have, so I am inclined to not have it.

**Comment by uscholdm** [comment]
@rjyounes Let's bring this to the group.

**Comment by stevenchalem** [comment]
It seems that the pattern has changed since this issue was opened. The restriction is now on the thing categorized.
```turtle
gist:Equipment
	a owl:Class ;
	owl:equivalentClass [
		a owl:Class ;
		owl:intersectionOf (
			gist:PhysicalIdentifiableItem
			[
				a owl:Restriction ;
				owl:onProperty gist:isCategorizedBy ;
				owl:someValuesFrom gist:EquipmentType ;
			]
		) ;
	] ;
	skos:definition "Human-made, tangible property other than land or buildings used to perform a task or activity."^^xsd:string ;
	skos:example "A machine, a router, a car, a baseball bat."^^xsd:string ;
	skos:prefLabel "Equipment"^^xsd:string ;
	.
```
Regardless, several category subclasses remain unaddressed.

**Comment by stevenchalem** [comment]
Some restrictions for consideration. I don't think these are all correct, but maybe they'll stimulate some good discussion.
```
ElectronicAddress ≡ Address ⊓ (isCategorizedBy some ElectronicAddressType)
PhysicalEvent ≡ Event ⊓ (isCategorizedBy some PhysicalActionType)
PhysicalAddress ≡ Address ⊓ (isCategorizedBy some PhysicalAddressType)
Address ≡ Content ⊓ (isCategorizedBy some AddressUsageType)
Aspect ≡ Thing ⊓ (isCategorizedBy some Discipline)

ElectronicAddressType ⊑ inverse(isCategorizedBy) only ElectronicAddress
PhysicalActionType ⊑ inverse(isCategorizedBy) only PhysicalEvent
PhysicalAddressType ⊑ inverse(isCategorizedBy) only PhysicalAddress
AddressUsageType ⊑ inverse(isCategorizedBy) only Address
Discipline ⊑ inverse(isCategorizedBy) only Aspect
```

**Comment by rjyounes** [comment]
I don't like the tight interconnection of `Aspect` and `Discipline` and it doesn't apply to all use cases. Otherwise I think the `allValuesFrom` restrictions are good. The `someValuesFrom` restrictions are incorrect, because you can have a category that does not yet categorize anything. Note that the `allValuesFrom` restrictions don't  violate our best practices because we generally want to infer something into the categorized class, not the category class.

**Comment by stevenchalem** [comment]
> I don't like the tight interconnection of `Aspect` and `Discipline` and it doesn't apply to all use cases. Otherwise I think the `allValuesFrom` restrictions are good. The `someValuesFrom` restrictions are incorrect, because you can have a category that does not yet categorize anything. Note that the `allValuesFrom` restrictions don't violate our best practices because we generally want to infer something into the categorized class, not the category class.

That leaves:
ElectronicAddressType ⊑ inverse(isCategorizedBy) only ElectronicAddress
PhysicalActionType ⊑ inverse(isCategorizedBy) only PhysicalEvent
PhysicalAddressType ⊑ inverse(isCategorizedBy) only PhysicalAddress
AddressUsageType ⊑ inverse(isCategorizedBy) only Address

**Comment by rjyounes** [comment]
The first 3 make sense to me.

The last is too strong, because it might categorize a relationship between an address and a person/org.

**Comment by stevenchalem** [comment]
> The `someValuesFrom` restrictions are incorrect, because you can have a category that does not yet categorize anything. 

@rjyounes Could you reconfirm this? gist already has things like
```turtle
gist:ProductSpecification
	a owl:Class ;
	owl:equivalentClass [
		a owl:Class ;
		owl:intersectionOf (
			gist:CatalogItem
			[
				a owl:Restriction ;
				owl:onProperty gist:isCategorizedBy ;
				owl:someValuesFrom gist:ProductCategory ;
			]
		) ;
	] ;
```

**Comment by rjyounes** [comment]
This is the other direction. A product specification must have a category, but a product category need not categorize anything.

**Comment by rjyounes** [comment]
> In the end, while correct, it is not really shedding much light, and its sort of obvious

@uschold The value is in being able to infer the type from being categorized with a certain type of category.

**Comment by rjyounes** [comment]
No clear benefit, and could cause problems down the road.
